### ISSUE TITLE
Upgrading to 2.3.5 breaks Swagger (Dotmailer API defintion)

### ISSUE THREAD

**Issue by moloughlin** [magento/magento2#28244]
### Preconditions (*)
   In composer.json:

```
"require": {
        "magento/magento-cloud-metapackage": ">=2.3.5 <2.3.6
},
```
   
Run `composer update --with-dependencies magento/magento-cloud-metapackage`

### Steps to reproduce (*)
Visit /swagger 

### Expected result (*)
Able to see Swagger docs

### Actual result (*)
Error shown: `Failed to load API definition.`
Error logged: `The "\DateTime" parameter type is invalid. Verify the parameter and try again.`

---

Set a breakpoint at vendor/magento/framework/Reflection/TypeProcessor.php:435
The problematic definition is \Dotdigitalgroup\Email\Api\Data\CouponAttributeInterface::getExpiresAt which declares a \DateTime return type, which the type processor seemingly can't handle.

**Comment by moloughlin** [comment]
@magento give me 2.3.5 instance

**Comment by magento-engcom-team** [comment]
Hi @moloughlin. Thank you for your request. I'm working on Magento 2.3.5 instance for you

**Comment by steved1982** [comment]
I am having the same issue here all SOAP requests now do not work on Magento 2.3.5

https://magento.stackexchange.com/questions/312864/magento-2-3-5-soap-datetime-is-invalid

**Comment by sta1r** [comment]
@moloughlin This was fixed in v4.5.1 of our connector: https://github.com/dotmailer/dotmailer-magento2-extension/pull/558

To resolve you can upgrade via:
```
composer require dotmailer/dotmailer-magento2-extension:"4.5.1 as 4.4.0"
php bin/magento setup:upgrade
```

or add a replace directive to remove our module if you prefer:
```
"replace": {  
        "dotmailer/dotmailer-magento2-extension": "*",  
}
```

**Comment by lano-vargas** [comment]
Same issue here, fresh installation 2.3.5 😒

**Comment by Emanuel-23** [comment]
I also confirm the issue and i am able to fix it by setting the composer alias as described by @sta1r for now. 

although this seems not right to me, instead we should update the version in all packages that currently depend on dotmailer prior to `dotmailer/dotmailer-magento2-extension` in version `4.5.1` (i.e. `magento/product-community-edition`)...

**Comment by Emanuel-23** [comment]
@magento-engcom-team
Can please someone check this, it seems not be solved in magento 2.3.6, although it looks like it could have been solved easily (see https://github.com/magento/magento2/issues/28244#issuecomment-630077997)

**Comment by engcom-Delta** [comment]
Hi @moloughlin @Emanuel-23 I'm not able to reproduce issue on clean 2.4-develop and on the latest composer version 2.4.1. Also was checked case with upgrade from 2.3.6 to 2.4.1 and swagger page is correctly displayed after upgrade
**Result:**
:heavy_check_mark: Swagger page is opened without errors
![image](https://user-images.githubusercontent.com/51681379/104764271-d2874980-576f-11eb-9aed-5d025c34df7e.png)


If you'd like to update the issue, please reopen it.
